src/ot_tracer.erl

```diff
@@ -31,27 +30,26 @@
 -callback start_span(opentelemetry:tracer(),
                      opentelemetry:span_name(),
                      ot_span:start_opts()) -> opentelemetry:span_ctx().
--callback with_span(opentelemetry:tracer(), opentelemetry:span_ctx()) -> ok.
--callback with_span(opentelemetry:tracer(), opentelemetry:span_ctx(), fun()) -> ok.
--callback end_span(opentelemetry:tracer()) -> ok.
+-callback set_span(opentelemetry:tracer(), opentelemetry:span_ctx()) -> ok.
+-callback end_span(opentelemetry:tracer(), opentelemetry:span_ctx()) -> ok.
 -callback current_span_ctx(opentelemetry:tracer()) -> opentelemetry:span_ctx().
 -callback span_module(opentelemetry:tracer()) -> module().
 
 -spec start_span(opentelemetry:tracer(), opentelemetry:span_name(),
                  ot_span:start_opts()) -> opentelemetry:span_ctx().
 start_span(Tracer={Module, _}, Name, Opts) ->
-    Module:start_span(Tracer, Name, Opts).
-
-with_span(Tracer={Module, _}, Span) when is_atom(Module) ->
-    Module:with_span(Tracer, Span).
-
--spec with_span(opentelemetry:tracer(), opentelemetry:span_ctx(), fun()) -> ok.
```

Comment:
`ot_tracer` needs a `with_span` that takes a function. Otherwise a user can't provide both a `Tracer` and a `fun`.

Reply:
I do not get what do you mean. Obviously they will not be able as such behaviour was moved one layer further to the `otel` module to simplify interface.